### PR TITLE
feat: realign toggle switch with design with reverse option

### PR DESCRIPTION
**Pull request by hqta**
# Objective
These changes aim to bring the ToggleSwitch in parity with the UI Kit and improve accessibility through amended color contrast. 

Changes made:
- On state changed to Green 400 and Green 500 for focus/hover state
- Increase contrast of Off state with Gray 400 with Gray 500 for hover/focus
- Add focus ring for consistency with other form elements
- Add controls story and design sheet story
- Reversed state
- Update storybook documentation

# Motivation and Context

Issue: https://cultureamp.atlassian.net/jira/software/projects/KDS/boards/289?selectedIssue=KDS-69

# Screenshots (if appropriate)
Design: https://www.figma.com/file/eZKEE5kXbEMY3lx84oz8iN/%E2%9D%A4%EF%B8%8F-UI-Kit%3A-Heart?node-id=14361%3A67850

# Checklist

- [ ] If this contains visual changes, has it been reviewed by a designer?
- [ ] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [ ] I have considered likely risks of these changes and got someone else to QA as appropriate
- [ ] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

**Review by jmitsios — APPROVED** [review]
Everything is great @hqta! Fine to approve.
I'm assuming the `freemium` description for the **Theme** prop will get deprecated as part of another separate task?

File: draft-packages/form/docs/ToggleSwitchField.stories.tsx [hunk 1]
```diff
@@ -72,191 +66,131 @@ export const OnKaizenSiteDemo = () => (
...
+    </div>
+    <div>
+      <Heading variant="heading-5" tag="h2" color="white">
+        Disabled
+      </Heading>
+    </div>
+    <ToggleStateContainer initialToggledStatus={ToggledStatus.ON}>
+      {({ toggledStatus, toggle }) => (
         <ToggleSwitchField
+          id="1"
```
Comment (HeartSquared) [inline draft-packages/form/docs/ToggleSwitchField.stories.tsx:155]: You'll want to change the ids for the Reversed stories. They trigger the Default ones when interacted with :)

**Comment by hqta** [comment]
> Everything is great @hqta! Fine to approve. I'm assuming the `freemium` description for the **Theme** prop will get deprecated as part of another separate task?

Yup that's correct - that's covered [here](https://github.com/cultureamp/kaizen-design-system/pull/2417)

**Comment by jmitsios** [comment]
@hqta Just thinking out loud - and we can have a proper discussion if needed.  In regards to the Labels, you can click them to activate the toggle, but hovering over the labels... should it activate the Toggle Switch's hover state?

**Comment by hqta** [comment]
> @hqta Just thinking out loud - and we can have a proper discussion if needed. In regards to the Labels, you can click them to activate the toggle, but hovering over the labels... should it activate the Toggle Switch's hover state?

I just had a quick look at the other form fields and the exhibit the same behaviour having a hover state. For consistency, I'd say keep the hover state.

**Review by christian314159 — DISMISSED** [review]
👏

**Review by christian314159 — DISMISSED** [review]

**Comment by HeartSquared** [comment]
@jmitsios @hqta
Another thing I noticed about hover states is that the cursor is `default` as opposed to `pointer`. Is that intended?

**Comment by hqta** [comment]
> @hqta Just thinking out loud - and we can have a proper discussion if needed. In regards to the Labels, you can click them to activate the toggle, but hovering over the labels... should it activate the Toggle Switch's hover state?

@dougmacknz @katsambarcus just wondering if you have any thoughts on this?

**Comment by dougmacknz** [comment]
> Just thinking out loud - and we can have a proper discussion if needed. In regards to the Labels, you can click them to activate the toggle, but hovering over the labels... should it activate the Toggle Switch's hover state?

That's a good spot! So our other form components trigger the hover styling when hovering the label, but the ToggleSwitch doesn't do this. @hqta I would look at adjusting this component to do the same as our others.

**Comment by NatalieLoyzaga** [comment]
LGTM :)

**Review by NatalieLoyzaga — APPROVED** [review]

**Review by jmitsios — APPROVED** [review]
Nicely done @hqta 👏

**Pull request merged**